### PR TITLE
feat(doc handle): documentation handling for bedrock

### PR DESCRIPTION
**Pull request by DenovVasil**
## Description
Added document handling for bedrock.

image

- documentation [PR](https://github.com/camunda/camunda-docs/pull/4795)
- e2e [PR](https://github.com/camunda/web-modeler/pull/12290)

- demo [link](https://camunda.slack.com/archives/C0350959WG1/p1734429689372359)

## Related issues

https://github.com/camunda/team-connectors/issues/969

### CODE REVIEW

File: connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java [hunk 1]
```diff
@@ -9,172 +9,210 @@
...
+                                        .modelId(this.modelId)
+                                        .messages(List.of(messageBuilder.build()))
+                                        .inferenceConfig(
+                                                config ->
+                                                        config
+                                                                .temperature(this.temperature)
+                                                                .maxTokens(this.maxTokens)
+                                                                .topP(this.topP)
+                                                                .build()));
+        String newMessage = converseResponse.output().message().content().getFirst().text();
```
Comment (DenovVasil) [inline connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java:127]: l

File: connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java [hunk 2]
```diff
@@ -116,39 +121,53 @@ public BedrockResponse execute(
...
+    List<BedrockContent> contentList =
+        new ArrayList<>(bedrockContentMapper.documentsToBedrockContent(this.newDocuments));
+    contentList.add(bedrockContentMapper.messageToBedrockContent(this.newMessage));
+
+    return new BedrockMessage(user, contentList);
+  }
+
+  private MessageMapper createMessageMapper() {
+    var bedrockContentMapper = new BedrockContentMapper(new DocumentMapper());
+    return new MessageMapper(bedrockContentMapper);
```
Comment (mathias-vandaele) [inline connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java:142]: So we are not returning the new Message and the history but only the history ? 
That requires changes in the documentation, and is it what is expected ?

File: connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java [hunk 3]
```diff
@@ -89,20 +90,24 @@ public final class ConverseData implements RequestData {
...
-            .map(
-                message ->
-                    messageBuilder
-                        .role(ConversationRole.valueOf(message.role()))
-                        .content(ContentBlock.fromText(message.message()))
-                        .build())
-            .toList();
+    var messageMapper = createMessageMapper();
+    var bedrockContentMapper = messageMapper.getBedrockContentMapper();
+    this.messagesHistory.add(prepareNewBedrockMessage(bedrockContentMapper));
```
Comment (mathias-vandaele) [inline connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java:107]: Having `prepareNewBedrockMessage` be part of MessageMapper would be clearer

File: connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java (hunk 2 above)
Comment (DenovVasil) [reply connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java:142]: Yes and Yes ))
I described this in the documentation [pr](https://github.com/camunda/camunda-docs/pull/4795)

File: connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java (hunk 3 above)
Comment (DenovVasil) [reply connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java:107]: make sense, I will do ))

Comment (DenovVasil) [reply connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java]: done

**Comment by Szik** [comment]
QA testing done.
findings:
- when defining a single document, the list-brakets "[]" are still needed or the following error is thown:
```
{"type":"io.camunda.connector.api.error.ConnectorException","message":"Cannot deserialize value of type `java.util.ArrayList<io.camunda.document.Document>` from Object value (token `JsonToken.START_OBJECT`)","code":"JSON_MISMATCH_ERROR"}
```
if this is intended, maybe this needs to be mentioned in the documentation
- a returned error by the connector does not result in an incident error in operate. is this intended?

**Comment by sbuettner** [comment]
@DenovVasil Can you make use of "Object" as the type and figure out whether its a single object document or multiple ones in the Connector?

**Comment by DenovVasil** [comment]
> QA testing done. findings:
> 
> * when defining a single document, the list-brakets "[]" are still needed or the following error is thown:
> 
> ```
> {"type":"io.camunda.connector.api.error.ConnectorException","message":"Cannot deserialize value of type `java.util.ArrayList<io.camunda.document.Document>` from Object value (token `JsonToken.START_OBJECT`)","code":"JSON_MISMATCH_ERROR"}
> ```
> 
> if this is intended, maybe this needs to be mentioned in the documentation
> 
> * a returned error by the connector does not result in an incident error in operate. is this intended?

We can provide a list of documents on the API bedrock, so for me this is expected behavior. 
Example:
1) Upload 2 documents via postman.
2)Make request with both files.
image
3) Get response 
image

**Comment by DenovVasil** [comment]
> QA testing done. findings:
> 
> * when defining a single document, the list-brakets "[]" are still needed or the following error is thown:
> 
> ```
> {"type":"io.camunda.connector.api.error.ConnectorException","message":"Cannot deserialize value of type `java.util.ArrayList<io.camunda.document.Document>` from Object value (token `JsonToken.START_OBJECT`)","code":"JSON_MISMATCH_ERROR"}
> ```
> 
> if this is intended, maybe this needs to be mentioned in the documentation
> 
> * a returned error by the connector does not result in an incident error in operate. is this intended?

The documentation states that the documents are a list.

**Pull request merged**